core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java

```diff
@@ -664,6 +666,133 @@ public void invalidDayOfYearArgument() {
+        Arguments.of("QUARTER", 1),
+        Arguments.of("YEAR", 2023),
+        Arguments.of("YEAR_MONTH", 202302)
+    );
+  }
+
+  @ParameterizedTest(name = "{0}")
+  @MethodSource({
+      "getDatetimeTestDataForExtractFunction",
+      "getTimeTestDataForExtractFunction",
+      "getDateTestDataForExtractFunction"})
+  public void testExtractWithDatetime(String part, long expected) {
+    lenient().when(nullRef.valueOf(env)).thenReturn(nullValue());
+    lenient().when(missingRef.valueOf(env)).thenReturn(missingValue());
+
+    FunctionExpression datetimeExpression = DSL.extract(
+        DSL.literal(part),
+        DSL.literal(new ExprDatetimeValue("2023-02-11 10:11:12.123")));
+
+    assertEquals(LONG, datetimeExpression.type());
+    assertEquals(expected, eval(datetimeExpression).longValue());
+    assertEquals(
+        String.format("extract(\"%s\", DATETIME '2023-02-11 10:11:12.123')", part),
+        datetimeExpression.toString());
+  }
+
+  private void datePartWithTimeArgQuery(String part, long expected) {
+    ExprTimeValue timeValue = new ExprTimeValue("10:11:12.123");
+
+    FunctionExpression datetimeExpression = DSL.extract(
+        functionProperties,
+        DSL.literal(part),
+        DSL.literal(timeValue));
+
+    assertEquals(LONG, datetimeExpression.type());
+    assertEquals(expected,
+        eval(datetimeExpression).longValue());
+  }
+
+
+  @Test
+  public void testExtractDatePartWithTimeType() {
+    lenient().when(nullRef.valueOf(env)).thenReturn(nullValue());
+    lenient().when(missingRef.valueOf(env)).thenReturn(missingValue());
+
+
+    datePartWithTimeArgQuery(
+        "DAY",
+        LocalDate.now(functionProperties.getQueryStartClock()).getDayOfMonth());
+
+    datePartWithTimeArgQuery(
+        "WEEK",
+        LocalDate.now(functionProperties.getQueryStartClock()).get(ALIGNED_WEEK_OF_YEAR));
+
+    datePartWithTimeArgQuery(
+        "MONTH",
+        LocalDate.now(functionProperties.getQueryStartClock()).getMonthValue());
+
+    datePartWithTimeArgQuery(
+        "YEAR",
+        LocalDate.now(functionProperties.getQueryStartClock()).getYear());
+  }
+
+  @ParameterizedTest(name = "{0}")
+  @MethodSource("getDateTestDataForExtractFunction")
+  public void testExtractWithDate(String part, long expected) {
+    lenient().when(nullRef.valueOf(env)).thenReturn(nullValue());
+    lenient().when(missingRef.valueOf(env)).thenReturn(missingValue());
+
+    FunctionExpression datetimeExpression = DSL.extract(
+        DSL.literal(part),
+        DSL.literal(new ExprDateValue("2023-02-11")));
+
+    assertEquals(LONG, datetimeExpression.type());
+    assertEquals(expected, eval(datetimeExpression).longValue());
+    assertEquals(
+        String.format("extract(\"%s\", DATE '2023-02-11')", part),
+        datetimeExpression.toString());
+  }
+
+  @ParameterizedTest(name = "{0}")
+  @MethodSource("getTimeTestDataForExtractFunction")
+  public void testExtractWithTime(String part, long expected) {
```

Comment:
Same as above is here (and in other tests too) - expected is given from outside the function, but function argument is defined inside.
Consider refactoring tests. For example, you can make a constant which stores this argument and define this constant near by expected values.